### PR TITLE
Quadrant counting example

### PR DESCRIPTION
**Pull request by ghoshgm**

### CODE REVIEW

File: example/count_quadrants/p4est_count_quadrants.c [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  p4est->user_pointer = &qs;
+  quadrant_count = count_quadrants (p4est);
+
+  /* Print quadrant count statistics for current rank. */
+  P4EST_VERBOSEF ("Local leaf quadrant count = %ld\n", qs.local_leaf_count);
+  P4EST_VERBOSEF ("Local branch quadrant count = %ld\n",
+                  qs.local_branch_count);
+
+  /* Print total quadrants in the tree structure. */
+  P4EST_VERBOSEF ("Global aggregrate quadrant count = %ld\n", quadrant_count);
```
Comment (cburstedde) [inline example/count_quadrants/p4est_count_quadrants.c:155]: Can this be changed to printing the global quadrant count (do we need an allreduce to compute it?) and the corresponding data member in p4est for comparison?

**Comment by cburstedde** [comment]
Thanks! Can we rename the directory to example/search/? And would it compile and run under CMake?

**Comment by cburstedde** [comment]
Hi @mkirilin, may I ask you to add these two programs to the CMake build? Thanks!

**Comment by mkirilin** [comment]
Hi @cburstedde, sure, I do it right once PR https://github.com/cburstedde/p4est/pull/231 is merged to avoid conflicts.

**Comment by cburstedde** [comment]
I'll merge this one now to continue with the proper directory name.
Please feel free afterwards to work on the CMake setup for examples.

**Pull request merged**